components/kyma-environment-broker/cmd/trialcleanup/main.go

```diff
@@ -132,32 +132,46 @@ func (s *TrialCleanupService) filterInstances(instances []internal.Instance, fil
 }
 
 func (s *TrialCleanupService) visitInstances(instances []internal.Instance, visit instanceVisitor) error {
+	var err error
+	var processedInstances int
+	var unprocessedInstances int
+	totalInstances := len(instances)
 	for _, instance := range instances {
-		err := visit(instance)
+		processed, err := visit(instance)
 		if err != nil {
-			return err
+			log.Error(errors.Wrap(err, "while sending expiration request"))
+			continue
+		}
+		if processed {
+			processedInstances += 1
+		} else {
+			unprocessedInstances += 1
 		}
 	}
-	return nil
+	failures := totalInstances - processedInstances - unprocessedInstances
+	log.Infof("To suspend: %+v processable: %+v unprocessable: %+v failures: %+v", totalInstances, processedInstances, unprocessedInstances, failures)
+	return err
 }
 
-func (s *TrialCleanupService) executeDryRun(instance internal.Instance) error {
+func (s *TrialCleanupService) executeDryRun(instance internal.Instance) (bool, error) {
 	log.Infof("instanceId: %+v createdAt: %+v (%.0f days ago) servicePlanID: %+v servicePlanName: %+v",
 		instance.InstanceID, instance.CreatedAt, time.Since(instance.CreatedAt).Hours()/24, instance.ServicePlanID, instance.ServicePlanName)
-	return nil
+	return true, nil
```

Comment:
for what is the boolean needed?

Reply:
In case of dry run this value is not necessary, could be discarded. In case of actual run it conveys information if instance was processable. We can have three outcomes - error, accepted request and request rejected as entity was unprocessable - it is needed to calculate summary.

Reply:
Changed - now this function returns nothing (void).